numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2434,39 +2467,51 @@ def launch_kernel(
     args,
     cooperative=False,
 ):
-    param_ptrs = [addressof(arg) for arg in args]
-    params = (c_void_p * len(param_ptrs))(*param_ptrs)
-
-    params_for_launch = addressof(params)
-    extra = 0
-
-    if cooperative:
-        driver.cuLaunchCooperativeKernel(
-            cufunc_handle,
-            gx,
-            gy,
-            gz,
-            bx,
-            by,
-            bz,
-            sharedmem,
-            hstream,
-            params_for_launch,
-        )
+    # Convert stream handle to cuda.core Stream object
+    if hstream is None:
+        hstream = 0
+    elif isinstance(hstream, binding.CUstream):
+        hstream = get_cuda_native_handle(hstream)
+    elif not isinstance(hstream, int):
+        hstream = 0
+    stream = ExperimentalStream.from_handle(hstream)
+
+    # Configure kernel launch parameters
+    config = LaunchConfig(
+        grid=(int(gx), int(gy), int(gz)),
+        block=(int(bx), int(by), int(bz)),
+        shmem_size=int(sharedmem) if sharedmem is not None else 0,
+        cooperative_launch=cooperative,
+    )
+
+    # Convert function handle to CUfunction binding type
+    if isinstance(cufunc_handle, binding.CUfunction):
+        cufunction = cufunc_handle
     else:
-        driver.cuLaunchKernel(
-            cufunc_handle,
-            gx,
-            gy,
-            gz,
-            bx,
-            by,
-            bz,
-            sharedmem,
-            hstream,
-            params_for_launch,
-            extra,
-        )
+        cufunction = binding.CUfunction(int(cufunc_handle))
+
+    # Convert arguments: cuda.core.launch expects numpy.uint64 for pointers,
+    # but can handle ctypes scalars directly
+    converted_args = [
+        np.uint64(arg.value if arg.value is not None else 0)
+        if isinstance(arg, c_void_p)
+        else arg
+        for arg in args
+    ]
+
+    # Create Kernel object and launch
+    # We use Kernel._from_obj with an ObjectCode stub since we're wrapping
+    # an existing CUfunction handle rather than creating from object code.
+    from cuda.core.experimental._module import _lazy_init
+
+    _lazy_init()
+    objcode_stub = object.__new__(ObjectCode)
+    kernel = Kernel._from_obj(cufunction, objcode_stub)
```

Comment:
I think this happens because again we unbox somewhere in the call site, and the box it back. Same question as above: If `launch_kernel` is not a public API, we should break it to avoid unboxing/boxing.

Reply:
I avoided breaking API in this change because I wanted to ensure that simply routing to cuda.core.launch didn't breaking any fundamental behaviour.  Then we could follow up with a PR that cleans up the boilerplate by breaking API.  Would you prefer to do that all in one PR?  @leofang @gmarkall